Tasks/AzureRmWebAppDeployment/AzureRMUtil.ts

```diff
@@ -0,0 +1,293 @@
+ 		 	auth : {
+ 		 		bearer : accessToken
+ 		 	}
+ 		 };
+
+ 		request (requestOptions, function (error, response, body) {
+ 		 	if ( error ) 
+ 		 		deferred.reject (error);
+ 		 	else {
+ 		 		if ( response.statusCode === 200 ) {
+ 		 			var obj = JSON.parse (body);
+ 		 			for ( var i = 0; i < Object.keys (obj.value).length; i++ ) {
+ 		 				if ( obj.value[i].name == webAppName ) {
+ 		 					deferred.resolve (obj.value[i]);
+						}
+					}
+ 		 			deferred.reject ('Error Occurred : No such web app');
+ 		 		}
+ 		 		else deferred.reject ('Error Occurred : [Staus Code : '+response.statusCode+']');
+
+ 		 	}
+ 		});
+ 		
+	}, function (error) {
+		taskLib.error (error);
+	});
+
+	return deferred.promise;
+
+};
+
+function getPublishProfileUtil ( SPN, webAppName: string, resourceGroupName: string, publishMethod: string, deployToSlotFlag: boolean, slotName: string ): Q.Promise<any> {
+
+    var deferred = Q.defer();
+    var insertInUrl = deployToSlotFlag ? "/slots/" + slotName : "";
+    getAuthorizationToken(SPN)
+        .then(function (accessToken) {
+        var requestOptions = {
+            url: 'https://management.azure.com/subscriptions/' + SPN.subscriptionId + '/resourceGroups/' + resourceGroupName + '/providers/Microsoft.Web/sites/' + webAppName + insertInUrl + '/publishxml?api-version=2015-08-01',
+            auth: {
+                bearer: accessToken
+            },
+            method: 'POST'
+        };
+        request(requestOptions, function (error, response, body) {
+            if (error)
+                deferred.reject(error);
+            else {
+                if (response.statusCode === 200) {
+                    parseString(body, function (error, result) {
+                        for (var i in result.publishData.publishProfile) {
+                            if (result.publishData.publishProfile[i].$.publishMethod === publishMethod)
+                                deferred.resolve(result.publishData.publishProfile[i].$);
+						}
+                        deferred.reject('Error : No Such Deploying Method Exists');
+                    });
+                }
+                else {
+                    deferred.reject('Error Fetching Deployment Publish Profile [Staus Code : '+response.statusCode+']');
+                }
+            }
+        });
+    }, function (error) {
+        taskLib.error(error);
+    });
+
+    return deferred.promise;
+}
+exports.getAzureRMWebAppPublishingProfileDetails = function (SPN, webAppName: string, resourceGroupName: string, publishMethod: string, deployToSlotFlag: boolean, slotName: string): Q.Promise<any> {
+    var deferred = Q.defer();
+      
+    if ( !deployToSlotFlag ) {
+        getAzureRMWebAppDetails_version2 ( SPN, webAppName, 'Microsoft.Web/sites' ).then ( function (webApp) {
+            resourceGroupName = webApp.id.split ('/')[4];
```

Comment:
this seems like a magic. Add comments on why you split and chose 4th item.
It would be great if you can move it into a small function named something like - fetch resourceGroupName from webAppId